src/guiChatConsole.cpp

```diff
@@ -611,7 +611,7 @@ bool GUIChatConsole::OnEvent(const SEvent& event)
 			prompt.nickCompletion(names, backwards);
 			return true;
 		}
-		else if(event.KeyInput.Char != 0 && !event.KeyInput.Control)
+		else if(isprint(event.KeyInput.Char) && !event.KeyInput.Control)
```

Comment:
I wonder which header is already including <ctype.h>

Anyhow, I don't really understand what this and the following lines with the casts are actually doing (not what you added, what's already there). 'KeyInput.Char' is of type 'wchar_t' isn't it? So, a) I don't know if isprint() is really meant to be used with something that's wchar_t; and b) I don't know what the lines below are doing.